docs/src/main/sphinx/release/release-389.md

```diff
@@ -0,0 +1,46 @@
+# Release 389 (6 July 2022)
+
+## General
+
+* Improve performance of queries which contain a large number of aggregations
+  that use the `row` type for representing intermediate state (e.g. `SUM`). ({issue}`12762`)
+* Improve join performance when spill to disk is not used. The old behavior for
```

Comment:
Not sure I understand ... spill to disk was not enabled by default in the past either..

Reply:
Currently spill to disk adds some overheads to join operator even if spill to disk is disabled just because the code is more complicated to handle the extra functionality. We're moving towards simplification of join operator by removing the spill to disk related code as this functionality is deprecated in favour of Tardigrade.
However, as an intermediate step we're keeping the old complicated join code which supports spill to disk around which creating a new simplified version of the code without spill to disk support. This is a temporary situation until Tardigrade is more mature.
Also, the new simplified join code will receive further optimisations which were not possible in the old code in near future.
The force-spilling-join-operator config is an escape hatch to allow falling back to old code in case of unforeseen issues.

Reply:
This should just be:
* Improve performance when spilling to disk is disabled.

> The force-spilling-join-operator config is an escape hatch to allow falling back to old code in case of unforeseen issues.

Not just an escape hatch -- it's needed if you do want to run with spilling enabled.

I haven't read that PR, but I don't understand why we need a separate config option vs piggybacking on the options to enable spilling. cc @raunaqmorarka